### DISCUSSION TITLE
Issues with Alt + Tab

### DISCUSSION THREAD

**Discussion by aksh1618** [linuxdeepin/developer-center#3060] [General | 常规讨论]
I'm using Deepin Desktop environment on Arch Linux. I like almost everything about it, except for the behavior of `Alt + Tab`. There are three things that I feel are 'wrong' with the current behavior:

- When you `Alt + Tab` to the desktop from an app *A*, and press `Alt+Tab` again, you get to another app *B* instead of *A*. I suspect this is because *A* is put on top of stack instead of the desktop, so `Alt + Tab` behaves like we are already on 'A'. when we are actually on the desktop. Also, it'd be a relief if the desktop would retain its position in the stack instead of moving to the end everytime.

- When a file clicked and dragged, `Alt + Tab` stops working exept for switching between the top two stack items (current and next). This limits the workflow, as this can be used to transfer files between apps, such as uploading an attachment from File Manager to Firefox. (Additionally, the file is 'removed' from the folder it was dragged from in File Manager, refreshing shows the file again.)

- Sometimes, when there is a dialog box open in a window (like settings dialog in an application), `Alt + Tab` behaves incorrectly. (This one is hard to reproduce, I think it happens when the dialog box isn't displayed as a different window)

**Comment by zccrs** [comment]
@sonald

**Comment by guilhermelowa** [comment]
Yes, I also like Deepin Desktop but find this behavior kinda annoying. Especially the first one. Any news on this?

**Comment by adziu** [comment]
Switching windows from keyboard is the biggest issue for me in Deepin.  Task switcher forgets the order, when windows are drawn **during** the switching operation.

Example:
1. I have three windows open: IDE, Browser and Chat. 
2. First, I use Browser, 
3. Then, I click with a mouse on IDE,
4. Now I want to go to Chat, 
    1. I press and keep Alt,
    2. I click Tab (most recently used Browser flashes), 
    3. I click Tab second time to get to Chat, 
    4. I release Alt. 
5. Now Alt+Tab should get me back to IDE,
6. Instead it gets me to Browser, because the act of drawing Browser brought it to the top of FIFO. 

Somewhat related issue:
https://github.com/linuxdeepin/developer-center/issues/1570

**Comment by meiyixiang** [comment]
We have recorded the problem you feedback, and it will take some time to analyze it
